### PR TITLE
Modularize Dumb For Cum quirk

### PR DESCRIPTION
**Pull request by LeDrascol**
# About The Pull Request
This PR modularizes and improves effects for the Dumb For Cum quirk.

The following changes are made:
- Adds comments
- Adds trait TRAIT_DUMB_CUM
- Adds mood quirk flag
- Adds quirk removal code
- Added penalty trait dumb
- Added penalty trait illiterate
- Added penalty trait pacifism
- Removes non-modular edits
- Renames trait TRAIT_DUMB4CUM to TRAIT_DUMB_CUM_CRAVE
- Removes useless return functions
- Updates flavor text
- Updates variable names
- Changes reagent process to check for trait before triggering uncrave
- Changes on_spawn event to add event

## Why It's Good For The Game
Fixes a large amount of non-modular edits, and improves code standards.

## A Port?
No.

## Pre-Merge Checklist
- [X] You tested this on a local server.
- [X] This code did not runtime during testing.
- [X] You documented all of your changes.

## Changelog
:cl:
fix: Dumb For Cum quirk can be added and removed mid-round
balance: Dumb For Cum quirk craving now causes pacifism and dumbness
/:cl:

### CODE REVIEW

**Comment by catrock31** [comment]
Hey man dumb for cum is supposed to make you illiterate I paid like 40$ for it

**Comment by LeDrascol** [comment]
> [...] dumb for cum is supposed to make you illiterate [...]

This PR has been updated to match the original bounty and PR.

File: modular_splurt/code/datums/traits/negative.dm [hunk 1]
```diff
@@ -125,42 +125,84 @@
...
-								  )
-	to_chat(quirk_holder, span_love("[pick(hungry_phrases)]"))
+/datum/quirk/dumb4cum/remove()
+	// Remove status trait
+	REMOVE_TRAIT(quirk_holder, TRAIT_DUMB_CUM_CRAVE, "dumb4cum")
+
+	// Remove penalty traits
+	ADD_TRAIT(quirk_holder, TRAIT_ILLITERATE, "dumb4cum")
+	ADD_TRAIT(quirk_holder, TRAIT_DUMB, "dumb4cum")
+	ADD_TRAIT(quirk_holder, TRAIT_PACIFISM, "dumb4cum")
```
Comment (BongaTheProto) [inline modular_splurt/code/datums/traits/negative.dm:149]: Wouldn't it be REMOVE_TRAIT ?

**Comment by BongaTheProto** [comment]
plz fix da conflicts

File: modular_splurt/code/datums/traits/negative.dm (hunk 1 above)
Comment (LeDrascol) [reply modular_splurt/code/datums/traits/negative.dm:149]: Thank you for pointing out this oversight. It will be addressed in a future commit.

**Pull request merged**